distributed/worker.py

```diff
@@ -1437,10 +1438,20 @@ async def start_unsafe(self):
         self.start_periodic_callbacks()
         return self
 
+    async def terminate(self, **kwargs):
+        return await self.close(nanny=True, **kwargs)
+
     @log_errors
     async def close(
-        self, report=True, timeout=30, nanny=True, executor_wait=True, safe=False
+        self,
+        timeout=30,
+        executor_wait=True,
+        nanny=False,
```

Comment:
⚠️ I'm concerned about flipping this default. Most of my comments here are around that. I think there are a number of places `close` is being used, without passing `nanny=True` explicitly (both in `worker.py` and in `scheduler.py`) and I am not convinced that they're all situations in which the worker process should be restarted if there's a nanny around.

I'm a little worried about this exacerbating problems with `Scheduler.remove_worker` https://github.com/dask/distributed/issues/6390, but I'm most concerned about verifying that it doesn't create an infinite restart loop.